### PR TITLE
Lint repo

### PR DESCRIPTION
**Pull request by caldwellst**
Set up linting for the repo and lint `main`. Based off discussion in #79 and the initial approach in #80.

### CODE REVIEW

File: .github/workflows/test-renv copy.yaml
Comment (hannahker) [inline .github/workflows/test-renv copy.yaml:1]: Update the file name here?

File: src/email/mailchimp/custom_segmentation.R [hunk 1]
```diff
@@ -85,52 +85,66 @@ mc_subscriber_emails <- function(df_ind, iso3, test) {
...
-              member$email_address
-            )
-          }
-        }
-        return(
-          NA_character_
-        )
-      }
-    )
+    member_emails <- interest_emails(interest = df_ind$mc_interest, geo_ids = geo_ids, test = test)
```
Comment (hannahker) [inline src/email/mailchimp/custom_segmentation.R:88]: Nice 👌

File: .lintr
Comment (hannahker) [inline .lintr:1]: Oh nice! Assuming this helps things work automatically with `styler`?

**Review by hannahker — APPROVED** [review]
On scan through this looks good to me! Do you think it's worth any further testing before merging? 

Just need to update the `yaml` file name though.

File: .lintr
Comment (caldwellst) [reply .lintr:1]: Yeah! Well for `{lintr}`, not `{styler}`. Automatically sets our preferences. We have to change a few things so it means if we run it locally we will get the same results as in the cloud.

**Comment by caldwellst** [comment]
Alright, merging! For now given the branch is passing the only test we have, the lint we implemented here, will merge. As we develop more and more tests, we can be more careful in our development but I'm prioritising speed at the moment hahaha.

**Pull request merged**